src/localplayer.cpp

```diff
@@ -438,15 +439,20 @@ void LocalPlayer::move(f32 dtime, Environment *env, f32 pos_max_d,
 	/*
 		Check properties of the node on which the player is standing
 	*/
-	const ContentFeatures &f = nodemgr->get(map->getNodeNoEx(m_standing_node));
+	const ContentFeatures &f_under = nodemgr->get(map->getNodeNoEx(m_standing_node));
+	const ContentFeatures &f_above = nodemgr->get(map->getNodeNoEx(
+		floatToInt(m_position + v3f(0, 0.2f * BS, 0), BS)));
+
 	// Determine if jumping is possible
-	m_can_jump = (touching_ground && !in_liquid && !is_climbing)
-			|| sneak_can_jump;
-	if (itemgroup_get(f.groups, "disable_jump"))
-		m_can_jump = false;
+	m_disable_jump = itemgroup_get(f_under.groups, "disable_jump");
+	m_can_jump = ((touching_ground && !is_climbing)
+			|| sneak_can_jump) && !m_disable_jump;
```

Comment:
Seems a possible change in behaviour in old_move (very dangerous), previously m_can_jump depended on in_liquid also.

Reply:
In liquids, the player usually does not touch ground. And when they do, it seems more realistic so that they can  push off the ground - also in liquids which don't allow moving up when floating.

Reply:
I don't think the realism of pushing off ground when in liquid is worth pursuing, liquid resistance will make such jumps pointless anyway. I'd rather not change this behaviour if it's not essential to the feature.

Reply:
If this isn't supported then I won't be able to use this feature to create a "poison gas" liquid that the player can move through as if it were air, as I had hoped. A player standing in non-breathable gas should still be able to jump normally if they're standing on a solid surface.

Reply:
To clarify i'm not necessarily opposed to this aspect, just unsure and being cautious.
Is it possible to make a liquid that the player moves as normal within, assuming jumping is possible within it?
Anyway i'm mostly neutral on this aspect.

Reply:
Pushing off from the ground will also be possible in regular liquids, causing overall a little faster elevation time (negligible). I know that it alters the current movement behaviour _slightly_ - compare it in a testing build if you think it would be concern,

Reply:
Ok, being able to jump in a liquid is probaly harmless, but needs testing.